README.md

```diff
@@ -1,60 +1,689 @@
+   - We also recommend settting up alerting systems to notify relevant stakeholders when events are sent to the DLQ, enabling prompt investigation.
+
+#### D. Simplified Listener Interface for Retry and DLQ
+
+To simplify the integration of retry and DLQ functionalities, **EventPeople** allows listeners to receive only essential parameters during event binding. The framework internally manages the instantiation and configuration of retry and DLQ components based on these parameters. If parameters are not provided, default configurations from the `Config` class are applied.
+
+**Listener Binding with Retry and DLQ:**
+
+```pseudo
+// Applying Retry and DLQ to a Listener with Custom Configurations
+Listener.on(
+    "order.order_service.updated.inventory_service",
+    callbackFunction,
+    maxAttempts = 5,
+    delayStrategy = "exponential", // Options: "fixed", "exponential"
+    dlqName = "order_service_dlq"
+)
+```
+
+**Interface Parameters:**
+
+- **`maxAttempts` (Integer):** Maximum number of retry attempts. Defaults to `Config.defaultRetryConfig.maxAttempts` if not provided.
+- **`delayStrategy` (String):** Strategy for delay between retries. Available options:
+  - `"fixed"`: Fixed delay between retries.
+  - `"exponential"`: Exponential backoff starting at 1 second (default).
+- **`dlqName` (String):** Name of the Dead Letter Queue. Defaults to the original queue name with `_dlq` suffix if not provided.
+
+**Default Exponential Backoff Configuration:**
+
+- **Initial Delay:** 1 second (1000 milliseconds)
+- **Multiplier:** 5 (delay  grows 5 times with each retry)
+- **Max Delay:** 10 minutes (600,000 milliseconds)
+
+**Usage Example:**
+
+```pseudo
+// Define callback
+function callbackFunction(event, context) {
+    // Process event
+    if (createUser(event.payload)) {
+        context.success()
+    } else {
+        context.fail()
+    }
+}
+
+// Subscribe to an event
+Listener.on(
+    "user.auth_service.created.all",
+    callbackFunction,
+    maxAttempts = 5,
+    delayStrategy = "exponential",
+    dlqName = "user_service_dlq"
+)
+```
+
+**Handling Retry Count and `isLastRetry` within Listeners:**
+
+While the retry logic is handled internally, listeners can still access the `retryCount` and `isLastRetry` attributes within the `Event` object to make informed decisions during event processing.
+
+```pseudo
+// Listening to an Event with Access to retryCount and isLastRetry
+Listener.on(
+    "user.auth_service.created.all",
+    function(event, context) {
+        // Optional: Customize behavior based on retry count
+        if (event.retryCount > 0) {
+            log("Retry attempt #" + event.retryCount + " for event: " + event.name)
+        }
+    
+        // Process event
+        createUser(event.payload)
+        context.success()
+    },
+    maxAttempts = 3,
+    delayStrategy = "exponential", // "fixed" or "exponential"
+    dlqName = "user_service_dlq"
+)
+```
+
+**Example with `isLastRetry`:**
+
+```pseudo
```

Comment:
It seems that using JavaScript (js) highlighting works quite well with this pseudocode. What do you think?

Reply:
This definition is not only to highlight the code but also to inform which language is being used. And as this repo is only the definition/guidelines for the lib implementation, we chose to go with `pseudo` as it is not an actual code/implementation.